Assets/Scripts/NPCs/NPCManager.cs

```diff
@@ -55,4 +90,63 @@ public float GetDistanceWithPlayer()
     {
         return Vector3.Distance(Indestructibles.Player.transform.position, transform.position);
     }
+
+    public bool IsChasing()
+    {
+        return _stateMachine.CurrentState.GetType() == typeof(Chase);
+    }
+    public void StartPunching()
+    {
+        if (_canPunch && !Punching)
+        {
+            Punching = true;
+            Animator.Play("Punch", PunchLayerIndex, 0f);
+            Animator.SetLayerWeight(PunchLayerIndex,1.0f); 
+        }
+    }
+
+    public void StopPunching()
+    {
+        Punching = false;
+    }
+    public void OnPlayerHit()
+    {
+        _stateMachine.StartWandering();
```

Comment:
After punching the player, I don't think the NPCs should start wandering since their goal is to catch the player. They should maybe have a small cooldown and restart chasing after.

Reply:
Disabled by default